Example/src/SharedElementTransition/Modals.tsx

```diff
@@ -0,0 +1,169 @@
+        goNext(nextScreen);
+      }}>
+      <Animated.View
+        style={
+          isOpen
+            ? { height: 500, marginTop: 50, backgroundColor: 'green' }
+            : { height: 120, marginTop: 20, backgroundColor: 'green' }
+        }
+        sharedTransitionTag={transitionTag + '1'}>
+        <Animated.Text
+          sharedTransitionTag={transitionTag + '2'}
+          style={{ width: '100%', height: 20 }}>
+          {title}
+        </Animated.Text>
+        <AnimatedImage
+          sharedTransitionTag={transitionTag + '3'}
+          source={photo}
+          style={{ width: '100%', height: isOpen ? 300 : 100 }}
+        />
+        <Animated.Text
+          sharedTransitionTag={transitionTag + '4'}
+          style={{ width: '100%', height: isOpen ? 100 : 0 }}>
+          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quas aliquid,
+          earum non, dignissimos fugit rerum exercitationem ab consequatur,
+          error animi veritatis delectus. Nostrum sapiente distinctio possimus
+          vel nam facilis ut?
+        </Animated.Text>
+      </Animated.View>
+    </TouchableNativeFeedback>
+  );
+}
+
+function Screen1({ navigation }: StackScreenProps<ParamListBase>) {
+  return (
+    <Animated.ScrollView style={{ flex: 1 }}>
+      {[...Array(6)].map((_, i) => (
+        <Card
+          key={i}
+          navigation={navigation}
+          title={'Title' + i}
+          transitionTag={'sharedTag' + i}
+          nextScreen="Screen2"
+        />
+      ))}
+    </Animated.ScrollView>
+  );
+}
+
+function Screen2({ route, navigation }: StackScreenProps<ParamListBase>) {
+  const { title, sharedTransitionTag } = route.params as any;
+
+  const goNext = () => {
+    navigation.navigate('Screen1', {
+      title,
+      sharedTransitionTag,
+    });
+  };
+
+  const translation = {
+    x: useSharedValue(0),
+    y: useSharedValue(0),
+  };
+  type AnimatedGHContext = {
+    startX: number;
+    startY: number;
+  };
+  const gestureHandler = useAnimatedGestureHandler<
+    PanGestureHandlerGestureEvent,
+    AnimatedGHContext
+  >({
+    onStart: (_, ctx) => {
+      ctx.startX = translation.x.value;
+      ctx.startY = translation.y.value;
+    },
+    onActive: (event, ctx) => {
+      translation.x.value = ctx.startX + event.translationX;
+      translation.y.value = ctx.startY + event.translationY;
+    },
+    onEnd: (_) => {
+      console.log(translation.x.value, translation.y.value);
+      if (Math.abs(translation.x.value) + Math.abs(translation.y.value) > 150) {
+        runOnJS(goNext)();
+      }
+      translation.x.value = withSpring(0);
+      translation.y.value = withSpring(0);
+    },
+  });
+
+  const animatedStyle = useAnimatedStyle(() => {
+    return {
+      transform: [
+        { translateX: translation.x.value },
+        { translateY: translation.y.value },
+        {
+          scale:
+            1 -
+            (Math.abs(translation.x.value) + Math.abs(translation.y.value)) /
```

Comment:
Can you make a derived value for distance and use it instead?

Reply:
Yes, but I don't think this necessary